### ISSUE TITLE
Does the code work when Batching Images?

### ISSUE THREAD

**Issue by JianHe0628** [geopavlakos/hamer#80]
Looking at the demo.py file, I see that there is an argument parser for batch_size, however it does not seem to be used in the code. I do see a batch size at the Dataloader area, but it takes in "dataset" which is created from a single img path. So, I am slightly confuse as to why the batch_size is set to 8 and whether the code support image frame batching? Apologies if I do have any misunderstandings with the code.

**Comment by JianHe0628** [comment]
Hi, oh yes I do get that the code is looping through the hands according to the batch_size at this section of the code

            # Render the result
            batch_size = batch['img'].shape[0]
            for n in range(batch_size):
                # Get filename from path img_path
                img_fn, _ = os.path.splitext(os.path.basename(img_path))
                
But my confusion was regarding what the code was looping through in the section below. I saw that the dataloader's batch_size is set at 8, so I went and printed out the enumerated count of the dataloader. I found that it would always only be 0 even when there are multiple hands.


        dataset = ViTDetDataset(model_cfg, img_cv2, boxes, right, rescale_factor=args.rescale_factor)
        dataloader = torch.utils.data.DataLoader(dataset, batch_size=8, shuffle=False, num_workers=0)

        all_verts = []
        all_cam_t = []
        all_right = []
        
        for batch in dataloader:
            batch = recursive_to(batch, device)
            with torch.no_grad():
                out = model(batch)

**Comment by geopavlakos** [comment]
I suspect you tried examples where there are many hands, but still less than 8. In an image like this:
https://github.com/shubham-goel/4D-Humans/blob/main/example_data/images/skates.png
the demo detects 17 hands and there are 3 batches overall.

**Comment by JianHe0628** [comment]
Ahh that makes perfect sense! Thank you for the replies
